scripts/build_renewable_profiles.py

```diff
@@ -263,21 +267,77 @@
     if snakemake.wildcards.technology.startswith("hydro"):
         country_shapes = gpd.read_file(paths.country_shapes)
         hydrobasins = gpd.read_file(resource["hydrobasins"])
+        ppls = load_powerplants(snakemake.input.powerplants)
+
+        hydro_ppls = ppls[ppls.carrier == "hydro"]
+
         # commented out as rivers may span across multiple countries
         # hydrobasins = hydrobasins[
         #     [
         #         any(country_shapes.geometry.intersects(geom))
         #         for geom in hydrobasins.geometry
         #     ]
         # ]  # exclude hydrobasins shapes that do not intersect the countries of interest
-        resource["plants"] = regions.rename(columns={"x": "lon", "y": "lat"}).loc[
+
+        resource["plants"] = regions.rename(
+            columns={"x": "lon", "y": "lat", "country": "countries"}
+        ).loc[
             [
                 # select busbar whose location (p) belongs to at least one hydrobasin geometry
-                any(hydrobasins.geometry.intersects(p))
-                for p in gpd.points_from_xy(regions.x, regions.y, crs=regions.crs)
+                # if extendable option is true, all buses are included
+                # otherwise only where hydro powerplants are available are considered
+                (
+                    (
+                        config.get("extendable", False)
+                        | (bus_id in hydro_ppls.bus.values)
+                    )
+                    & any(hydrobasins.geometry.intersects(p))
+                )
+                for (p, bus_id) in zip(
+                    gpd.points_from_xy(regions.x, regions.y, crs=regions.crs),
+                    regions.index,
+                )
```

Comment:
This filtering is a bit confusing/complex. :D I know it's compact but hard to maintain. Do you have ideas how to simplify it?

One idea: 
- put the list comprehension out of .loc statement
- ... maybe you have some more ideas

Reply:
Maybe two parentheses were too many, but I liked to move the list comprehension outside.
Done option 1.